apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/RequestResponsePopulateMdcFilterTest.java

```diff
@@ -0,0 +1,127 @@
+package gov.cms.bfd.server.war;
+
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.slf4j.spi.MDCAdapter;
+
+/** Unit tests for {@link BfdMDC}. */
+@ExtendWith(MockitoExtension.class)
+public final class RequestResponsePopulateMdcFilterTest {
+  /** A mocked MDC adapter for use in the tests. */
+  @Mock MDCAdapter mdcMock;
+
+  @Mock HttpServletRequest servletRequest;
+  @Mock HttpServletResponse servletResponse;
+  @Mock FilterChain filterChain;
+
+  /** Set up tests with a mocked MDC adapter and make sure the MDC context is clear. */
+  @BeforeEach
+  public void beforeEach() {
+    mdcMock.clear();
+    BfdMDC.setMDCAdapter(mdcMock);
```

Comment:
Unfortunately the MDC adapter is a static variable which interferes with other tests when setting the mock value here. I'll have to think about the best workaround for this.